### PR TITLE
NXmx: handle multidimensional arrays

### PR DESCRIPTION
**Pull request by phyy-nx**
Data in NeXus can be 3 or 4 dimensional:
3D: Nimages by slow by fast
4D: Nimages by Nmodules by slow fast

Slice image_size and reshape the raw_data in these cases

Adapted from FormatNexus.py:
https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/nexus.py#L852-L857
https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/nexus.py#L1329-L1331

### CODE REVIEW

**Comment by phyy-nx** [comment]
I think this line needs to be worked on as well, but I don't have a use case for it right now:
https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/nexus/__init__.py#L478

**Comment by phyy-nx** [comment]
> I think this line needs to be worked on as well, but I don't have a use case for it right now: https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/nexus/__init__.py#L478

Fixed!  I have a usecase for a 3D mask now

File: src/dxtbx/nexus/__init__.py [hunk 1]
```diff
@@ -404,9 +404,14 @@ def equipment_component_key(dependency):
...
-        image_size = cast(Tuple[int, int], tuple(map(int, module.data_size[::-1])))
-        assert len(image_size) == 2
+        # is stored in module.data_size. Additionally, data_size can have more than 2
+        # dimensions, for multi-module detectors. So take the last two dimensions and reverse
+        # them.  Examples:
+        # [1,2,3,4][-1:-3:-1] --> [4, 3]
+        # [1,2,3][-1:-3:-1] --> [3, 2]
+        # [1,2][-1:-3:-1] --> [2, 1]
+        image_size = cast(Tuple[int, int], tuple(map(int, module.data_size[-1:-3:-1])))
+        assert len(image_size) == 2, image_size
```
Comment (ndevenish) [inline src/dxtbx/nexus/__init__.py:414]:
```suggestion
        # [1,2,3,4] --> [4, 3]
        # [1,2,3]   --> [3, 2]
        # [1,2]     --> [2, 1]
        image_size = (int(module.data_size[-1]), int(module.data_size[-2])
```
I think this is more explicit, and doesn't involve reading ranged negative slices (it perhaps should have been this before anyway)

**Comment by ndevenish** [comment]
Do you have any test files/files we can make in the right shape, to add to the testing?

**Comment by rjgildea** [comment]
@phyy-nx I've attempted to add a basic test for what I understand of this use case. Could you take a look and verify that it makes sense from your perspective?

**Review by rjgildea — APPROVED** [review]
This adds support for a use case that wasn't previously handled, LGTM 👍

**Comment by phyy-nx** [comment]
Hi @ndevenish I reviewed @rjgildea's test and it looks perfect.  I think this is ok to merge as is.

**Pull request merged**